Add how-to of happy path for Lakehouse sync DOCS-604 (#5740)
EnterpriseDB/docs: advocacy_docs/edb-postgres-ai/analytics/how_to_lakehouse_sync.mdx

---
title: Lakehouse Sync
navTitle: Lakehouse Sync

> description: How-to guide for performing a Lakehouse Sync.

Reviewer: ```suggestion
description: How to perform a Lakehouse Sync.
```

---

## Overview

A Lakehouse Sync is a way to capture information from a transactional database at a point in time and sync that information to a Managed Store Location in an AWS region.

Once synced, the data in the MSL is organized into Lakehouse tables. These Lakehouse tables can then be queried by a Lakehouse node optimized for higher-performance queries using a vectorized query engine for Lakehouse tables.

## Perfroming a Lakehouse Sync

### Prerequisites

> - a Postgres cluster running on your EDB Postgres AI Cloud Service account

Reviewer: Since this is the first mention of the product, we can add the trademark symbol
```suggestion
- a Postgres cluster running on your EDB Postgres AI® Cloud Service account
```

> - a Postgres cluster running on your EDB Postgres AI Cloud Service account

Author: > Is this hosted or managed?

both right? You must have a BigAnimal instance.

### Navigate to Lakehouse Sync

1. Login to your EDB Postgres AI Console.

> 2. From the landing page, select the project with the database instance you want to sync. If it is not shown on the landing page, select the **View Projects** link in the **Projects** section and select your project from there.

Author: Yes it is just the landing page, or at least that's what I remember Jace saying.

> 2. From the landing page, select the project with the database instance you want to sync. If it is not shown on the landing page, select the **View Projects** link in the **Projects** section and select your project from there.

Reviewer 2: Yes I personally think it's easier to tell people to "go to the cluster page for the transactional cluster you want to sync, click 'quick actions,' and 'Sync to Lakehouse' (this is what DJ shows in his screenshot)."

The Project -> Migrations -> Create Migration path is actually not the ideal one, especially because when you start a migration and need to select from a number of sources, it's not super obvious from the dropdown which source is which...

3. Select the **Migrate** dropdown in the left navigation bar and then select **Migrations**.

4. Select the **Create New Migration** button.

### Define Lakehouse Sync

5. Give the sync a **Name**, then select a **Source Cluster** and the **Database** you want to sync.

> 6. If you have already created a MSL you want to use, select that MSL from the list of available MSLs and skip to step 10. If not, select the **Create New Managed Storage Location** button to bring up the **Add Managed Storage Location** modal.

Reviewer: Not sure if these are the correct terms, but I would change the wording a bit so non-native speakers can easily follow as well. 
```suggestion
6. If you have already created a MSL you want to use, select that MSL from the list of available MSLs and skip to step 10. If not, select the **Create New Managed Storage Location** button to open the **Add Managed Storage Location** pop-up window.
```

> 7. Select the radio button next to the AWS region in which you want to create your new MSL.

Author: got it.

> 8. Define a location prefix in the form near the bottom of the modal to complete the definition of the MSL.

Reviewer: Instead of modal or dialogue, maybe pop-up window? Might be a more "visual" term.

> 8. Define a location prefix in the form near the bottom of the modal to complete the definition of the MSL.

Author: I am going to try dialog. Pop-up window just makes me think of pop-up adds.

> 8. Define a location prefix in the form near the bottom of the modal to complete the definition of the MSL.

Reviewer 3: Unfortunately it's not a pop-up (no window frame).

> 9. Select the **Create Managed Storage Location** button.

Author: I let them know in #beacon-ux

> 9. Select the **Create Managed Storage Location** button.

Reviewer 3: Apparently yes. :)

> 9. Select the **Create Managed Storage Location** button.

Reviewer 2: We're changing to say "Create" in both places.

> 10. Select the **Tables** tab next to the **Get Started** tab near the top of the page and select which tables and columns you wish to be included in the migration.

Author: I gave it a go.

---
title: Lakehouse Sync
navTitle: Lakehouse Sync
description: How to perform a Lakehouse Sync.
deepToC: true
---

## Overview

Performing a Lakehouse Sync is a way to capture information from a transactional database at a point in time and sync that information to a Managed Store Location (MSL).

The Lakehouse sync process organizes the transactional database data into Lakehouse tables stored in the MSL. This process allows the data to be queried by a Lakehouse node, which is optimized for higher-performance queries using a vectorized query engine designed for Lakehouse tables.

## Performing a Lakehouse Sync

### Prerequisites

> - a Postgres cluster hosted and managed by EDB Postgres AI® Cloud Service

Reviewer: I thing it is Services, but I am not sure how we are using the product name. 
```suggestion
- a Postgres cluster hosted and managed by EDB Postgres AI® Cloud Services
```

Author: Looks like it is Cloud Service from this: https://www.enterprisedb.com/products/edb-postgres-ai-cloud-service

4. Select the **Create New Migration** button.

### Define Lakehouse Sync

5. Give the sync a **Name**, then select a **Source Cluster** and the **Database** you want to sync.

6. If you have already created an MSL you want to use, select that MSL from the list of available MSLs and move on to [Selecting Tables](#selecting-tables) below. If not, select the **Create New Managed Storage Location** button to open the **Add Managed Storage Location** dialog.

7. Select the AWS region for the new MSL.

8. Set a location prefix in the form near the bottom of the **Add Managed Storage Location** dialog to complete the definition of the MSL. A location prefix is a unique name used to identify any resources and assets associated with the MSL.

![List of MSLs](./images/msl_list.png)

9. Select the **Create Managed Storage Location** button.

### Selecting Tables

10. Select the **Tables** tab next to the **Get Started** tab near the top of the page and select which tables and columns you wish to be included in the migration.

### Start Lakehouse Sync

11. Select the **Start Lakehouse Sync** button.

> 12. If successful, you will see your Lakehouse sync with the 'Creating' status under 'MOST RECENT' migrations on the Migrations page. The time taken to perform a sync can depend upon how much data is being synchronized and may take several hours. 

Reviewer 2: The first sync in a project will take significantly longer (at least 1 hour) due to provisioning Kafka infrastructure. This also applies when you are performing a sync after a long time without one, because we shut down idle Kafka infrastructure (in fact, it might be after 24 hours of non-use.... not sure.)

Author: Okay I added an admonition regarding the infrastructure provisioning time the first sync of a project.
